internal/service/currency-service_test.go

```diff
@@ -0,0 +1,178 @@
+package service_test
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"genesis-currency-api/internal/service"
+	"genesis-currency-api/pkg/config"
+	myerrors "genesis-currency-api/pkg/errors"
+	"github.com/stretchr/testify/suite"
+)
+
+type CurrencyServiceSuite struct {
+	suite.Suite
+	sut service.CurrencyServiceInterface
+}
+
+func TestCurrencyServiceImplSuite(t *testing.T) {
+	suite.Run(t, &CurrencyServiceSuite{})
+}
+
+func (suite *CurrencyServiceSuite) TestGetCurrencyInfo_checkResult() {
+	// SETUP
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/external-api" {
+			suite.Failf("Expected to request '/fixedvalue', got: %s", r.URL.Path)
+		}
+		w.WriteHeader(http.StatusOK)
+
+		_, err := w.Write([]byte(`[{"ccy":"USD","base_ccy":"UAH","buy":"39.95","sale":"40.87"}]`))
+		if err != nil {
+			return
+		}
+	}))
```

Comment:
Thx! I though about it, but quickly realized that it would be an overkill, because it is 'sort' of natural go API for http mocking and it is how it was ment to be used. And there are many changes for each test, so no functions needed.